sdk/loadtestservice/Azure.Developer.LoadTesting/src/Models/Test.cs

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core.Json;
+
+namespace Azure.Developer.LoadTesting.Models
+{
+    /// <summary> Load test model. </summary>
+    public partial class Test
+    {
+        private MutableJsonElement _element;
+
+        /// <summary> Initializes a new instance of Test. </summary>
+        public Test()
+        {
+            // TODO: Store a way to make an empty MJD - avoid the ToArray() allocation below.
+            _element = MutableJsonDocument.Parse(BinaryData.FromBytes("{}"u8.ToArray())).RootElement;
+
+            // TODO: If we are suggesting a wholesale replacement of ChangeTrackingDictionary,
+            // we will need to confirm that MJD provides the complete set of features this type provides.
+        }
+
+        internal Test(MutableJsonElement element)
+        {
+            _element = element;
+
+            if (_element.TryGetProperty("passFailCriteria", out MutableJsonElement passFailCriteriaElement))
+            {
+                _passFailCriteria = new PassFailCriteria(passFailCriteriaElement);
+            }
+
+            if (_element.TryGetProperty("secrets", out MutableJsonElement secretsElement))
+            {
+                _secrets = new MutableJsonDictionary<Secret>(secretsElement);
+            }
+
+            if (_element.TryGetProperty("environmentVariables", out MutableJsonElement environmentVariablesElement))
+            {
+                _environmentVariables = new MutableJsonDictionary<string>(environmentVariablesElement);
+            }
+        }
+
+        private PassFailCriteria _passFailCriteria;
+        /// <summary> Pass fail criteria for a test. </summary>
+        public PassFailCriteria PassFailCriteria
+        {
+            get
+            {
+                if (_passFailCriteria == null)
+                {
+                    _element.SetProperty("passFailCriteria", new { });
+                    _passFailCriteria = new PassFailCriteria(_element.GetProperty("passFailCriteria"));
+                }
+
+                return _passFailCriteria;
+            }
+
+            // TODO: need to test what happens if caller sets this, how the interaction
+            // across MJEs in the MJD works.
+            set { _passFailCriteria = value; }
+        }
```

Comment:
Is such logic applied to all the properties or just patch impacted properties?

Reply:
My understanding of PATCH APIs (although please correct me if I'm wrong), is that any property on a model could be updated by a PATCH operation.  In this implementation, I've used a private field and corresponding property for any JSON element that is an object or an array in the JSON.  Elements that don't have children didn't appear to need this, since we could use TryGetProperty on the MutableJsonElement to get and set them.  An example of this is in PassFailMetric, where there are string and number elements.

Reply:
Yeah if a model is impacted then every property should be impacted. But what I'm asking is what if the model is not impacted. For example, I just have `@get op operation(): Model;`. No patch operation.

Should it be like:
```C#
public partial class Model
{
        public Model(bool property)
        {
            Property = property;
        }

        /// <summary> Property. </summary>
        public bool Property { get; set; }
}
```
or
```C#
public partial class Model
{
        public Model()
        {
        }

        public bool? Property
        {
            get
            {
                if (!_element.TryGetProperty("property", out MutableJsonElement value))
                {
                    return null;
                }

                if (!value.TryGetBool(out bool d))
                {
                    return null;
                }

                return d;
            }

            set => _element.SetProperty("property", value);
        }
}
```

Reply:
If your answer is former, then what the Model looks like when we have
```
@get op operation(): Model;
@patch op operation2(model: Model): void;
```
If your answer is latter, then the required property changes to optional.

Reply:
The model will be the same regardless of whether it is part of a patch operation or not.  The properties on the model in this PR are optional because the service's API spec described them as optional, not because they are part of a patch operation.

Reply:
Yeah, but what if the properties in the spec are required? We should change it to optional because every property in patch model is optional, right?

Reply:
I don't think we should.  The model should reflect what is in the API spec.  We will use the same model for normal operations and for PATCH operations.